### PR TITLE
feat: support online state and excluding feature names in dump json model

### PR DESCRIPTION
**Pull request by jackgerrits**

### CODE REVIEW

File: vowpalwabbit/global_data.cc [hunk 1]
```diff
@@ -257,17 +260,47 @@ std::string dump_weights_to_json_weight_typed(
...
+          adaptive_value = rapidjson::kNullType;
+          normalized_value = current_weight_state[1];
+        }
+        if (parameter_holder.adaptive && parameter_holder.normalized)
+        {
+          adaptive_value = current_weight_state[1];
+          normalized_value = current_weight_state[2];
+        }
+
+        extra_state_value.AddMember("adaptive", adaptive_value, allocator);
```
Comment (ataymano) [inline vowpalwabbit/global_data.cc:300]: is it possible to make gd.cc responsible for adding this since this is gd-specific?

Comment (jackgerrits) [reply vowpalwabbit/global_data.cc:300]: Yeah I agree it would be good. I am not sure of a good way to make that happen at the moment to be honest.

Comment (ataymano) [reply vowpalwabbit/global_data.cc:300]: I see - looks like the fact that we have adaptive/normalized field as array_parameters members is hack as well.
Does it mean that if we try to output state with non-gd learner, we just not going to see extra output for now?

Comment (jackgerrits) [reply vowpalwabbit/global_data.cc:300]: The dump_json_weights feature explicitly restricts usage to GD for now. I figured it was better to restrict usage until we work out a way to better support other use cases.

**Review by ataymano — APPROVED** [review]

**Pull request merged**